source/core/CEntryPoints.cpp

```diff
@@ -194,20 +194,20 @@ VIREO_EXPORT const char* EggShell_ReadValueString(TypeManagerRef tm,
     return "";
 }
 //------------------------------------------------------------
-//! Get the Length of an Array Symbol. Returns -1 if the Symbol is not found or not a 1D Array.
-VIREO_EXPORT Int32 EggShell_Get1DArrayLength(TypeManagerRef tm, const char* viName, const char* eltName)
+//! Get the Length of a dimension in an Array Symbol. Returns -1 if the Symbol is not found or not an Array.
+VIREO_EXPORT Int32 EggShell_GetArrayDimLength(TypeManagerRef tm, const char* viName, const char* eltName, Int32 dim) 
 {
     TypeManagerScope scope(tm);
     void *pData = null;
 
     SubString objectName(viName);
     SubString path(eltName);
     TypeRef actualType = tm->GetObjectElementAddressFromPath(&objectName, &path, &pData, true);
-    if (actualType == null || !actualType->IsArray() || actualType->Rank() != 1)
+    if (actualType == null || !actualType->IsArray())
```

Comment:
Instead of getting rid of the Rank() test, you should probably change it to:
dim >= actualType->Rank() 

If GetLength() gets an out of range dim, it asserts.